app/src/main/java/pl/jakubokrasa/kursakademiaandroida/features/character/data/CharacterCached.kt

```diff
@@ -0,0 +1,55 @@
+package pl.jakubokrasa.kursakademiaandroida.features.character.data
+
+import androidx.room.Entity
+import com.google.gson.annotations.SerializedName
+import pl.jakubokrasa.kursakademiaandroida.core.data.api.model.location.LocationRemote
+import pl.jakubokrasa.kursakademiaandroida.core.data.api.model.origin.OriginResponse
+import pl.jakubokrasa.kursakademiaandroida.core.data.local.model.OriginCached
+import pl.jakubokrasa.kursakademiaandroida.features.character.domain.model.Character
+import pl.jakubokrasa.kursakademiaandroida.features.location.data.local.model.LocationCached
+
+@Entity
+data class CharacterCached(
+    val id: Int,
+    val episode: List<String>,
+    val gender: String,
+    val image: String,
+    val location: LocationCached,
```

Comment:
Żeby móc wrzucić obiekt zawierający inny obiekt do bazy danych konieczne będzie skorzystanie z adnotacji `@Embeded`. W tym przypadku konieczne będzie też ustawienie prefixu, który pozwoli rozróżnić pola tych obiektów na poziomie bazy danych. Przykład jak może wyglądać taki model znajdziesz tutaj: https://github.com/keresmi/Android-Clean-Architecture-Sample/blob/master/app/src/main/java/pl/akademiaandroida/android_clean_architecture_sample/features/characters/data/local/model/CharacterCached.kt.

Reply:
Ok,  a dlaczego w tym linku CharacterLocationCached ma tylko 2 pola? LocationCached ma nie tylko name i url. Czy to nie pomyłka?

![image](https://user-images.githubusercontent.com/35777139/109394688-86abf080-7928-11eb-94e6-42b748624985.png)

Reply:
@keresmi